### ISSUE TITLE
task updates not working

### ISSUE THREAD

**Issue by rubycut** [influxdata/kapacitor#941]
I am updating existing tasks using PATCH httpd request, but it seems that update doesn't work properly, for example if I add new httpdOut node, it is not displayed, when fetched with browser.

Also alerts don't trigger.

I am using kapacitor 1.0.0

For now, I am just deleting tasks and recreating them again, then everything works.

**Comment by nathanielc** [comment]
@rubycut Can you provide a simple script using curl that reproduces the issue?

**Comment by rubycut** [comment]
If task is in weird state, which means it exists in /kapacitor/v1/tasks/task_name, but it doesn't exist in /kapacitor/v1/tasks, then I can't delete it, and I can't recreate it.

**Comment by rubycut** [comment]
So, kapacitor cli first executes GET request "GET /kapacitor/v1/tasks?dot-view=attributes&fields=link&limit=100&offset=0&pattern=ping&replay-id=&script-format=formatted" to see if task is present, so, since empty response is returned, it never executes DELETE request.

**Comment by rubycut** [comment]
And even doing DELETE request manually with curl, I need to do it twice, it's stuck somewhere:

```
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|stream->stream0] 2016/09/27 17:24:53 D! closing c: 1625 e: 95
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|stats2->derivative4] 2016/09/27 17:24:53 D! closing c: 1560 e: 1560
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|derivative4->window5] 2016/09/27 17:24:53 D! closing c: 1536 e: 1536
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|window5->http_out6] 2016/09/27 17:24:53 D! closing c: 744 e: 744
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|http_out6->mean7] 2016/09/27 17:24:53 D! closing c: 744 e: 744
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|mean7->http_out8] 2016/09/27 17:24:53 D! closing c: 744 e: 744
Sep 27 17:24:53 xxx kapacitord[16602]: [edge:ping|http_out8->alert9] 2016/09/27 17:24:53 D! closing c: 744 e: 744
Sep 27 17:24:53 xxx kapacitord[16602]: [task_store] 2016/09/27 17:24:53 D! task ping finished
Sep 27 17:24:53 xxx kapacitord[16602]: [task_master:main] 2016/09/27 17:24:53 E! Stopped task: ping mean12: invalid influxql func mean with field percent_packet_lost: invalid field type: <nil>
Sep 27 17:24:53 xxx kapacitord[16602]: [task_store] 2016/09/27 17:24:53 E! task ping finished with error: mean12: invalid influxql func mean with field percent_packet_lost: invalid field type: <nil>
Sep 27 17:24:53 xxx kapacitord[16602]: [httpd] 31.192.122.38 - - [27/Sep/2016:17:24:53 +0000] "DELETE /kapacitor/v1/tasks/ping HTTP/1.1" 204 0 "-" "curl/7.38.0" 4d52980a-84d7-11e6-99ee-000000000000 145555
```

**Comment by nathanielc** [comment]
@rubycut Thanks, I have a few ideas as to what might be happening at this point.

**Comment by rubycut** [comment]
Yes, I don't see any point in updating tasks, just drop them and recreate every time.

**Comment by Artemmkin** [comment]
@rubycut kapacitor remembers the state of an alert, so there is a very good point in updating tasks. For instance, what happens if you've got an alert triggered, and you remove and recreate the task?   You will either have a new alert triggered or the old alert never being reset.  According to [this](https://docs.influxdata.com/kapacitor/v1.5/troubleshooting/frequently-asked-questions/#is-the-alert-state-and-alert-data-lost-happen-when-updating-a-script), I believe the update operation should keep state of the last triggered alert.

**Comment by Artemmkin** [comment]
I guess the current workaround is to [disable/enable the task](https://github.com/influxdata/kapacitor/issues/1510) for this to work.

**Comment by rubycut** [comment]
@Artemmkin , you have a point there, deleting and recreating tasks will trigger alerts again. 

Also, many tasks store some data in window property, so when task is deleted, this window needs to be filled again.
